object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```
@@ -279,6 +306,276 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
+              || succRegularStore.getValue((LocalVariableNode) (assignmentNode.getTarget()))
+                  == null) {
+
+            if (nodes.size() == 0) { // If the cur block is special or conditional block
+              reportAlwaysCallErrors(assignmentNode, succRegularStore, null);
+
+            } else { // If the cur block is Exception/Regular block then it checks AlwaysCall
+              // condition in the store right after the last node
+              Node last = nodes.get(nodes.size() - 1);
+              CFStore storeAfter = getStoreAfter(last);
+              reportAlwaysCallErrors(
+                  assignmentNode,
+                  storeAfter,
+                  (last instanceof AssignmentNode) ? getAnnotatedType(last.getTree()) : null);
+            }
+          }
+
+          if (succRegularStore.getValue((LocalVariableNode) (assignmentNode.getTarget())) == null) {
+            // If the lhs of the assignmentNode isn't in the successor's input store, it means the
+            // variable is going out of scope
+            toRemove.add(assignmentNode);
+          }
+        }
+
+        newDefs.removeAll(toRemove);
+        propagate(Pair.of(succ, newDefs), visited, worklist);
+      }
+    }
+  }
+
+  /**
+   * it inputs a set of AssignmentNodes(defs) and a LocalVariableNode(localVariableNode), if there
+   * is an AssignmentNode in defs that its lhs is equal to localVarNode, then it returns that
+   * AssignmentNode, otherwise it returns null.
+   *
+   * @param defs
+   * @param localVarNode
+   * @return
+   */
+  private AssignmentNode getAssignmentNodeOfVar(
+      Set<AssignmentNode> defs, LocalVariableNode localVarNode) {
+
+    for (AssignmentNode assignmentNode : defs) {
+      LocalVariableNode lhs = (LocalVariableNode) assignmentNode.getTarget();
+
+      if (lhs.getElement() == localVarNode.getElement()) {
+        return assignmentNode;
+      }
+    }
+
+    return null;
```

Here I think you should throw an exception rather than just returning `null`.  We only expect this method to be called when a relevant assignment is in the `defs` set